### DISCUSSION TITLE
Is this unit test correctly implemented (should get a list of users by alpha (username, email) (fuzzy match))?

### DISCUSSION THREAD

**Discussion by wlsf82** [cypress-io/cypress-realworld-app#760] [Q&A]
I was exploring the unit tests of the RWA, and I realized that [this test](https://github.com/cypress-io/cypress-realworld-app/blob/develop/src/__tests__/users.test.ts#L40) describes that it should get a list of users by alpha (username, email) (fuzzy match), but, the implementation is searching by `firstName` instead (see https://github.com/cypress-io/cypress-realworld-app/blob/develop/src/__tests__/users.test.ts#L42.)

I just wanted to confirm if this is correct?

**Comment by wlsf82** [comment]
What about this one https://github.com/cypress-io/cypress-realworld-app/blob/develop/src/utils/__tests__/transactionUtils.test.ts#L134?

The previous ones on the same file that test `getting query without something` all have two expectations: passing some data + the status, and a second expectation only passing the status.

Shouldn't this be the case of [this](https://github.com/cypress-io/cypress-realworld-app/blob/develop/src/utils/__tests__/transactionUtils.test.ts#L134) test too? I mean, shouldn't we have another expectation passing only the status here too?

**Comment by kevinold** [comment]
Hi @wlsf82,

Moved this over to a discussion.  I'm certain the unit tests could be updated and feel free to play around with them and submit a PR if you arrive at corrections or additional assertions that could be added.  

Our goal with these unit tests was to provide a minimum set of foundational unit tests when querying data from the lowdb database, since we do not have the features of a full ORM.

With this minimal set of unit tests in place, we were able to round out testing coverage using end-to-end tests.

For overall coverage of searching, you can refer to this Cypress test which covers this area of code from an end-to-end perspective - https://github.com/cypress-io/cypress-realworld-app/blob/18f3b23835618c853d043f457b05e948d9715d3f/cypress/tests/ui/new-transaction.spec.ts#L244

Hope this helps.

**Reply by wlsf82** [reply]
Thanks, @kevinold!
